experimental/list_formatter/src/lib.rs

```diff
@@ -49,6 +49,9 @@
 extern crate alloc;
 
 mod list_formatter;
+#[cfg(feature = "provider_transform_internals")]
+pub mod provider;
+#[cfg(not(feature = "provider_transform_internals"))]
```

Comment:
Issue: Although I would like `mod provider` to be private, it is used in public trait bounds.  The compiler usually complains about this, but I've seen cases where it doesn't detect a private type used in a public signature.  I think the macro might be obscuring this fact.  CC @Manishearth